api/src/main/java/org/hawkular/apm/api/services/Criteria.java

```diff
@@ -139,9 +140,9 @@ public Criteria setEndTime(long endTime) {
      */
     public long calculateEndTime() {
         if (endTime == 0) {
-            return System.currentTimeMillis();
+            return TimeUnit.NANOSECONDS.toMicros(System.nanoTime());
```

Comment:
We can't use nanoTime as a replacement for currentTimeMillis - which is a true timestamp. As described in the nanoTime docs `This method can only be used to measure elapsed time and is not related to any other notion of system or wall-clock time. `.